docs/_newsfragments/api_class.deprecation.rst

```diff
@@ -0,0 +1 @@
+The ``falcon.API`` class is deprecated, and will be removed in the next major release 4.0, use ``falcon.App`` instead.
```

Comment:
This could benefit from some cosmetic touch-ups:
* `.deprecation.rst` is not among the [extensions supported by Towncrier](https://github.com/hawkowl/towncrier#news-fragments).
* No newline and of file